src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -57,6 +58,12 @@ object Checking {
           val bounds = tparams.map(tparam =>
             tparam.info.asSeenFrom(tycon.tpe.normalizedPrefix, tparam.owner.owner).bounds)
           checkBounds(args, bounds, _.substDealias(tparams, _))
+        case Select(qual, name) if name.isTypeName =>
+          checkRealizable(qual.tpe, qual.pos)
+        case SelectFromTypeTree(qual, name) if name.isTypeName =>
+          checkRealizable(qual.tpe, qual.pos)
+        case SingletonTypeTree(ref) =>
+          checkRealizable(ref.tpe, ref.pos)
         case _ =>
```

Comment:
That's not how things are done in DOT. We have one scheme to ensure soundness that we know works and a lot of others that we thought worked at some time, but didn't in the end. So we should stick to the one that works. I believe approximating realizability of other types is too strict and is also not necessary.